packages/ra-ui-materialui/src/list/datagrid/Datagrid.tsx

```diff
@@ -200,7 +220,24 @@ export const Datagrid: React.ForwardRefExoticComponent<
         [data, isRowSelectable, onSelect, onToggleItem, selectedIds]
     );
 
-    if (isPending === true) {
+    const accessibleChildren = useMemo(() => {
```

Comment:
This may break the `optimized` variant

Reply:
What optimized variant?

Reply:
`<Datagrid optimized>`, which memoizes the render regardless of changes in the children